crates/mapdiffbot2/src/job_processor.rs

```diff
@@ -73,98 +73,128 @@ fn render(
     );
 
     //do removed maps
-    let removed_directory = format!("{}/r", out_dir.display());
-    let removed_directory = Path::new(&removed_directory);
+    let removed_directory = out_dir.to_path_buf().join("r");
+    let removed_directory = removed_directory.as_path();
+
     let removed_errors = Default::default();
 
     let removed_maps = with_checkout(&base_branch, repo, || {
         let maps = load_maps_with_whole_map_regions(removed_files, &path)
             .context("Loading removed maps")?;
         render_map_regions(
             &base_context,
-            &maps.iter().collect::<Vec<_>>(),
+            maps.iter()
+                .map(|(k, v)| (k.as_str(), v))
+                .collect::<Vec<_>>()
+                .as_slice(),
             &base_render_passes,
-            removed_directory,
+            (removed_directory, blob_client.clone()),
             "removed.png",
             &removed_errors,
+            crate::rendering::MapType::Base,
         )
         .context("Rendering removed maps")?;
         Ok(maps)
     })?;
 
     //do added maps
-    let added_directory = format!("{}/a", out_dir.display());
-    let added_directory = Path::new(&added_directory);
+    let added_directory = out_dir.to_path_buf().join("a");
+    let added_directory = added_directory.as_path();
+
     let added_errors = Default::default();
 
     let added_maps = with_checkout(&head_branch, repo, || {
         let maps =
             load_maps_with_whole_map_regions(added_files, &path).context("Loading added maps")?;
         render_map_regions(
             &head_context,
-            &maps.iter().collect::<Vec<_>>(),
+            maps.iter()
+                .map(|(k, v)| (k.as_str(), v))
+                .collect::<Vec<_>>()
+                .as_slice(),
```

Comment:
this feels like it might be an unnecessary collect, maybe render_map_regions should take an iterator? I think there's a pattern for this kind of thing but I can't for the life of me remember it